llama_poe/poe_api/base_handler.py

```diff
@@ -0,0 +1,118 @@
+import json
+import logging
+from typing import AsyncIterable, Dict, Optional, Union
+
+from fastapi.responses import JSONResponse
+from sse_starlette.sse import ServerSentEvent
+
+from poe_api.types import (AddDocumentsRequest, ContentType, QueryRequest,
```

Comment:
If possible, it would be better to depend on `fastapi_poe` (which is already on PyPI) so that users can take advantage of new features and fixes in that package as we improve it. This looks to be mostly a copy of the `fastapi_poe` code.

Reply:
Is your comment here specific to the data models in types, or also the API and handler code?

The reason I replicated the API and base handler code is because I wanted to expose a `add_document` endpoint. Any advice/preference on how I should do this?

Reply:
You could simply inherit from `PoeHandler` and add the `add_document` method.

Reply:
That way you can still avoid duplicating the code for the handler though. For example, there's helper functions for the different event types. If we add another one, right now your code would need manual changes to pull it in. You can still provide your own `run` function.

Reply:
Sure, will revise to avoid replicating the BaseHandler!

Reply:
Revised! Would really appreciate another quick review!